lib/puppet/settings/alias_setting.rb

```diff
@@ -0,0 +1,33 @@
+class Puppet::Settings::AliasSetting
+  attr_reader :name, :alias_name
+
+  def initialize(args = {})
+    @name = args[:name].to_s
+    @alias_name = args[:alias_for]
+    @alias_for = Puppet.settings.setting(alias_name)
+  end
+
+  def optparse_args
+    if short
+      ["--#{name}", "-#{short}", desc, :REQUIRED]
+    else
+      ["--#{name}", desc, :REQUIRED]
+    end
+  end
```

Comment:
This might need to take into consideration other settings types that have their own `optparse_args`, eg: https://github.com/puppetlabs/puppet/blob/main/lib/puppet/settings/boolean_setting.rb#L12-L18